### PR TITLE
zed dev compile: Add describe flag

### PR DESCRIPTION
**Pull request by mattnibs**
The commit adds a describe flag to the compile command the outputs the JSON output of compiler/describe.

### CODE REVIEW

File: cmd/zed/dev/compile/command.go [hunk 1]
```diff
@@ -200,6 +210,20 @@ func (c *Command) writeDAG(seq dag.Seq) {
 	}
 }
 
+func (c *Command) writeDescribe(ctx context.Context, seq dag.Seq, lk *lake.Root) {
+	info, err := describe.Analyze(ctx, seq, data.NewSource(nil, lk), nil)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	out, err := json.MarshalIndent(info, "", "  ")
```
Comment (nwt) [inline cmd/zed/dev/compile/command.go:219]: Nit: Be consistent with `normalize`.
```suggestion
	out, err := json.MarshalIndent(info, "", "    ")
```

File: compiler/job.go [hunk 2]
```diff
@@ -199,3 +201,18 @@ func VectorFilterCompile(rctx *runtime.Context, query string, src *data.Source,
...
+	}
+	entry, err := semantic.Analyze(ctx, seq, src, head)
+	if err != nil {
+		if list, ok := err.(parser.ErrorList); ok {
+			list.SetSourceSet(sset)
+		}
+		return nil, err
+	}
+	return describe.Analyze(ctx, entry, src, head)
+}
```
Comment (nwt) [inline compiler/job.go:218]: I don't see any benefit in putting this here so how about keeping it as compiler/describe.Analyze and adding compiler/describe.AnalyzeDAG?

Comment (mattnibs) [reply compiler/job.go:218]: sure

File: service/handlers.go [hunk 3]
```diff
@@ -179,7 +179,7 @@ func handleQueryDescribe(c *Core, w *ResponseWriter, r *Request) {
 		return
 	}
 	src := data.NewSource(storage.NewRemoteEngine(), c.root)
-	info, err := describe.Analyze(r.Context(), req.Query, src, &req.Head)
+	info, err := describe.Analyze(r.Context(), req.Query, nil, src, &req.Head)
```
Comment (nwt) [inline service/handlers.go:182]: Nit: Why add this argument if you're just going to pass nil?

**Review by nwt — APPROVED** [review]

**Pull request merged**